src/book/eventsFetcher.service.js

```diff
@@ -0,0 +1,114 @@
+/* eslint-disable class-methods-use-this */
+
+const moment = require('moment');
+const { reduce } = require('asyncro');
+
+const { errors } = require('./book.errors');
+const { redisEventsKey } = require('../../config/book.config.json');
+const { assertOrThrow } = require('../../utils');
+
+class EventsFetcherService {
+  constructor(clientExtractor, roomResources, redisClient, usersMapper) {
+    this.clientExtractor = clientExtractor;
+    this.roomResources = roomResources;
+    this.redis = redisClient;
+    this.usersMapper = usersMapper;
+  }
+
+  getRoomEmails() {
+    const roomEmails = [];
+    this.roomResources.forEach((room) => {
+      roomEmails.push(room.email);
+    });
+
+    return roomEmails;
+  }
+
+  minutesToStart(date) {
+    const now = moment();
+    const end = moment(date);
+    const duration = moment.duration(end.diff(now));
+    const diffMins = Math.round(duration.asMinutes());
+
+    return diffMins;
+  }
+
+  async getAllEvents() {
+    const client = await this.getClient();
+    const res = await client.events.list({
+      calendarId: process.env.BOT_EMAIL,
+      timeMin: moment(),
+      timeMax: moment().add(20, 'minutes').toDate(),
+      singleEvents: true,
+    });
+
+    assertOrThrow(res, errors.GOOGLE_API_ERROR);
+
+    return res.data.items;
+  }
+
+  containsRoom(attendees, roomEmails) {
+    return attendees.some(attendee => roomEmails.includes(attendee.email));
+  }
+
+  async filterUpcomingWithResources(events, roomEmails) {
+    const filtered = events.filter((event) => {
+      const minutesToStart = this.minutesToStart(event.start.dateTime);
+      return (minutesToStart > 0 && this.containsRoom(event.attendees, roomEmails));
+    });
+
+    return filtered;
+  }
+
+  async mapEvent(event) {
+    let res;
+    try {
+      res = await this.usersMapper.getMappedSlackUsers([event.creator.email]);
+    } catch (err) {
+      throw err;
+    }
+
+    return JSON.stringify({
+      [event.creator.email]: {
+        ownerId: res[event.creator.email],
+        name: event.summary,
+        url: event.htmlLink,
+        confirmed: false,
+        requested: false,
+        dateTime: event.start.dateTime,
+      },
+    });
+  }
+
+  async saveToDatabase(events) {
+    let redisResult;
+
+    const added = await reduce(events, async (accumulator = [], event) => {
+      redisResult = await this.redis.hget(redisEventsKey, event.id);
```

Comment:
You can remove the `let redisResult` and make it `const` here.
```suggestion
      const redisResult = await this.redis.hget(redisEventsKey, event.id);
```

Also you can just return the result of the reduce, no need of `const added; return added;`

Reply:
changed